staging/src/k8s.io/apimachinery/pkg/util/clock/clock.go

```diff
@@ -291,18 +291,19 @@ func (f *fakeTimer) C() <-chan time.Time {
 func (f *fakeTimer) Stop() bool {
 	f.fakeClock.lock.Lock()
 	defer f.fakeClock.lock.Unlock()
-
+	fired := true
```

Comment:
I prefer `fired := false` here and setting `fired = true` on line 298 if you find the waiter channel. That would simplify the return statement (no negation necessary).

Reply:
I am not sure I understand.  `Stop()` is defined to return `true` if the timer has _not_ fired.  If we want `Stop()` to return the value of a variable without negation then that variable should be named something like "notFired" --- and we still have a negation but it is in the variable name rather than the logic.  In my opinion that makes the logic more difficult to read.  Also, changing line 298 to ignore `w.fired` strikes me as a brittle coding even though it looks correct at the moment.

Reply:
It seems that either the destChan is there, in which case the fired attribute is false, or if it has fired then it shouldn't actually be present so our equality check won't occur. Why should we have a layer of indirection here?

Reply:
No, actually, `w.fired` might be true so we positively need line 298 to read it.